packages/core-ethereum/src/hashedSecret.ts

```diff
@@ -1,120 +1,93 @@
-import type HoprEthereum from '.'
 import { Hash } from './types'
 import Debug from 'debug'
-const log = Debug('hopr-core-ethereum:hashedSecret')
 import { randomBytes } from 'crypto'
 import { u8aToHex, u8aConcat, iterateHash, recoverIteratedHash } from '@hoprnet/hopr-utils'
 import type { Intermediate } from '@hoprnet/hopr-utils'
-
 import { publicKeyConvert } from 'secp256k1'
+import { hash as hashFunctionUtils, waitForConfirmation } from './utils'
+import { OnChainSecret, OnChainSecretIntermediary } from './dbKeys'
+import type { LevelUp } from 'levelup'
+import type { HoprChannels } from './tsc/web3/HoprChannels'
+import type Account from './account'
 
 export const DB_ITERATION_BLOCK_SIZE = 10000
 export const TOTAL_ITERATIONS = 100000
 export const HASHED_SECRET_WIDTH = 27
 
+const log = Debug('hopr-core-ethereum:hashedSecret')
 const isNullAccount = (a: string) => a == null || ['0', '0x', '0x'.padEnd(66, '0')].includes(a)
 
-class HashedSecret {
-  constructor(private coreConnector: HoprEthereum) {}
+export async function hashFunction(msg: Uint8Array): Promise<Uint8Array> {
+  return (await hashFunctionUtils(msg)).slice(0, HASHED_SECRET_WIDTH)
+}
 
-  /**
-   * @returns a promise that resolves to a Hash if secret is found
-   */
-  private async getOffChainSecret(): Promise<Hash | undefined> {
-    try {
-      return await this.coreConnector.db.get(Buffer.from(this.coreConnector.dbKeys.OnChainSecret()))
-    } catch (err) {
-      if (!err.notFound) {
-        throw err
-      }
-      return
+async function getFromDB<T>(db: LevelUp, key): Promise<T | undefined> {
+  try {
+    return await db.get(Buffer.from(key))
+  } catch (err) {
+    if (!err.notFound) {
+      throw err
     }
+    return
   }
+}
+
+class HashedSecret {
+  private initialized: boolean = false
+  private onChainSecret: Hash
+  private offChainSecret: Hash
+
+  constructor(private db: LevelUp, private account: Account, private channels: HoprChannels) {}
 
   /**
    * @returns a deterministic secret that is used in debug mode
    */
   private async getDebugAccountSecret(): Promise<Hash> {
-    const account = await this.coreConnector.hoprChannels.methods
-      .accounts((await this.coreConnector.account.address).toHex())
-      .call()
-
+    const account = await this.channels.methods.accounts((await this.account.address).toHex()).call()
     return new Hash(
-      (
-        await this.coreConnector.utils.hash(
-          u8aConcat(new Uint8Array([parseInt(account.counter)]), this.coreConnector.account.keys.onChain.pubKey)
-        )
-      ).slice(0, HASHED_SECRET_WIDTH)
+      await hashFunction(u8aConcat(new Uint8Array([parseInt(account.counter)]), this.account.keys.onChain.pubKey))
     )
   }
 
-  public async hashFunction(msg: Uint8Array): Promise<Uint8Array> {
-    let toHash = msg
-
-    // @TODO Uncomment this to have salted hashes
-    // let toHash = u8aConcat(new TextEncoder().encode('HOPRnet'), await this.coreConnector.account.address, msg)
```

Comment:
salted hashes are supposed to come in with the next smart contract upgrade. Please keep the line in, referring to it in https://github.com/hoprnet/hoprnet/issues/1257

Reply:
Future code should live in PRs not comments.

Reply:
@robertkiel can u make sure the ticket related to salt is reopened?

Reply:
https://github.com/hoprnet/hoprnet/issues/1257